src/Esfa.Vacancy.Register.Api/App_Start/AutoMapperConfig.cs

```diff
@@ -10,10 +11,45 @@ public static void Configure()
         {
             Mapper.Initialize(cfg =>
             {
+                cfg.CreateMap<string, int?>().ConvertUsing(new StringToNullIntConverter());
+                cfg.CreateMap<int, ApiTypes.VacancyType>().ConvertUsing(new IntToEnumConverter<ApiTypes.VacancyType>());
+                cfg.CreateMap<int, ApiTypes.WageUnit>().ConvertUsing(new IntToEnumConverter<ApiTypes.WageUnit>());
                 cfg.CreateMap<DomainTypes.Vacancy, ApiTypes.Vacancy>()
-                .ForMember(apiType => apiType.VacancyType, opt => opt.MapFrom(source => (ApiTypes.VacancyType)source.VacancyTypeId))
-                .ForMember(apiType => apiType.WageUnit, opt => opt.MapFrom(source => (ApiTypes.WageUnit)source.WageUnitId));
+                    .ForMember(apiType => apiType.VacancyType, opt => opt.MapFrom(source => source.VacancyTypeId))
+                    .ForMember(apiType => apiType.WageUnit, opt => opt.MapFrom(source => source.WageUnitId));
+
             });
         }
     }
+
+    public class StringToNullIntConverter : ITypeConverter<string, int?>
+    {
+        public int? Convert(string source, int? destination, ResolutionContext context)
+        {
+            int i;
+            if ((string.IsNullOrWhiteSpace(source)) || (!int.TryParse(source, out i)))
+                return null;
+            return i;
+        }
+    }
+
+    public class IntToEnumConverter<T> : ITypeConverter<int, T>
```

Comment:
Unfortunately Enum cannot be used as constraint. As usual John Skeet has come up with a [solution ](https://codeblog.jonskeet.uk/2009/09/10/generic-constraints-for-enums-and-delegates/) but I think it is too much for what we want to achieve.